File: CS/DXExample.Module/HidePivotGridFieldsController.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.PivotChart;

namespace DXExample.Module {
    public class HidePivotGridFieldsController : ObjectViewController<DetailView, IAnalysisInfo> {
        protected override void OnActivated() {
            base.OnActivated();
            foreach (AnalysisEditorBase editor in View.GetItems<AnalysisEditorBase>()) {
                editor.ControlCreated += editor_ControlCreated;
            }
        }
        void editor_ControlCreated(object sender, EventArgs e) {
            ISupportPivotGridFieldBuilder supportFB = ((AnalysisEditorBase)sender).Control as ISupportPivotGridFieldBuilder;
            if (supportFB != null) {
                supportFB.FieldBuilder = new CustomPivotFieldBuilder((IAnalysisControl)supportFB);
                supportFB.FieldBuilder.SetModel(Application.Model);
            }
        }
        protected override void OnDeactivated() {
            base.OnDeactivated();
            foreach (AnalysisEditorBase editor in View.GetItems<AnalysisEditorBase>()) {
                editor.ControlCreated -= editor_ControlCreated;
            }
        }
    }
}

File: VB/DXExample.Module/HidePivotGridFieldsController.vb
Imports System

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.PivotChart

Namespace DXExample.Module
    Public Class HidePivotGridFieldsController
        Inherits ObjectViewController(Of DetailView, IAnalysisInfo)

        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            For Each editor As AnalysisEditorBase In View.GetItems(Of AnalysisEditorBase)()
                AddHandler editor.ControlCreated, AddressOf editor_ControlCreated
            Next editor
        End Sub
        Private Sub editor_ControlCreated(ByVal sender As Object, ByVal e As EventArgs)
            Dim supportFB As ISupportPivotGridFieldBuilder = TryCast(DirectCast(sender, AnalysisEditorBase).Control, ISupportPivotGridFieldBuilder)
            If supportFB IsNot Nothing Then
                supportFB.FieldBuilder = New CustomPivotFieldBuilder(DirectCast(supportFB, IAnalysisControl))
                supportFB.FieldBuilder.SetModel(Application.Model)
            End If
        End Sub
        Protected Overrides Sub OnDeactivated()
            MyBase.OnDeactivated()
            For Each editor As AnalysisEditorBase In View.GetItems(Of AnalysisEditorBase)()
                RemoveHandler editor.ControlCreated, AddressOf editor_ControlCreated
            Next editor
        End Sub
    End Class
End Namespace
